### PR TITLE
ASV: dt64arr_to_periodarr

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: asv_bench/benchmarks/tslibs/period.py [hunk 1]
```diff
@@ -104,3 +112,19 @@ def setup(self, size, freq):
...
+        _tzs,
+    ]
+    param_names = ["size", "freq", "tz"]
+
+    def setup(self, size, freq, tz):
+        arr = np.arange(10, dtype="i8").repeat(size // 10)
+        self.i8values = arr
+
+    def time_dt64arr_to_periodarr(self, size, freq, tz):
+        dt64arr_to_periodarr(self.i8values, freq, tz)
```
Comment (jorisvandenbossche) [inline asv_bench/benchmarks/tslibs/period.py:130]: Is it possible to time the public function/method that in the end uses this instead?

Comment (jbrockmendel) [reply asv_bench/benchmarks/tslibs/period.py:130]: its possible, yes.  but ATM im aiming for thorough coverage of tslibs in isolation, and PRs that dont touch tslibs can skip any of their asvs

Comment (jreback) [reply asv_bench/benchmarks/tslibs/period.py:132]: we likely *already* have coverage at a higher level, these fine-grained asv's are useful.

**Comment by jreback** [comment]
checks are failing......

**Pull request merged**